packages/web-components/src/tabs/tabs.ts

```diff
@@ -0,0 +1,194 @@
+import { attr } from '@microsoft/fast-element';
+import { FASTTabs } from '@microsoft/fast-foundation';
+import { Tab } from '../index.js';
+import { TabsAppearance, TabsSize } from './tabs.options.js';
+
+export const TabTokenNames = {
+  tabIndicatorOffset: '--tabIndicatorOffsetX',
+  tabIndicatorScale: '--tabIndicatorScaleX',
+} as const;
+
+export interface TabData {
+  id: string;
+  x: number;
+  y: number;
+  height: number;
+  width: number;
+}
+
+/**
+ * TabList extends FASTTabs and is used for constructing a fluent-tab-list custom html element.
+ *
+ * @class TabList component
+ * @public
+ */
+export class Tabs extends FASTTabs {
+  /**
+   * activeTabData
+   * The positional coordinates and size dimensions of the active tab. Used for calculating the offset and scale of the tab active indicator.
+   */
+  private activeTabData: TabData = { id: '', x: 0, y: 0, height: 0, width: 0 };
+  /**
+   * previousActiveTabData
+   * The positional coordinates and size dimensions of the active tab. Used for calculating the offset and scale of the tab active indicator.
+   */
+  private previousActiveTabData: TabData = { id: '', x: 0, y: 0, height: 0, width: 0 };
+  /**
+   * activeTabOffset
+   * Used to position the active indicator for animations of the active indicator on active tab changes.
+   */
+  private activeTabOffset = 0;
+  /**
+   * activeTabScale
+   * Used to scale the tab active indicator up or down as animations of the active indicator occur.
+   */
+  private activeTabScale = 1;
+
+  /**
+   * appearance
+   * There are two modes of appearance: transparent and subtle.
+   */
+  @attr
+  public appearance?: TabsAppearance = TabsAppearance.transparent;
+
+  /**
+   * disabled
+   * Used for disabling all click and keyboard events for the tabs, child tab elements and tab panel elements. UI styling of content and tabs will appear as "grayed out."
+   */
+  @attr({ mode: 'boolean' })
+  public disabled?: boolean;
+
+  /**
+   * size
+   * Used to set the size of all the tab controls, which effects text size and margins. Three sizes: small, medium and large.
+   */
+  @attr
+  public size?: TabsSize = 'medium';
```

Comment:
A couple notes here - if we're setting a value in code we should always use the actual value from the const object we've created. With that said, we should be setting these via CSS only and using tsdoc comments to enumerate what the default is. Setting this will reflect it back to the DOM which will clutter.